### PR TITLE
Use borrowed Arena in GethTraceBuilder

### PR DESCRIPTION
**Pull request by nkysg**
Closes https://github.com/paradigmxyz/revm-inspectors/issues/65

### CODE REVIEW

File: src/tracing/builder/geth.rs [hunk 1]
```diff
@@ -11,20 +11,26 @@ use alloy_rpc_types::trace::geth::{
...
+    collections::{BTreeMap, HashMap, VecDeque},
+};
 
 /// A type for creating geth style traces
 #[derive(Clone, Debug)]
-pub struct GethTraceBuilder {
+pub struct GethTraceBuilder<'a> {
     /// Recorded trace nodes.
-    nodes: Vec<CallTraceNode>,
+    nodes: Cow<'a, Vec<CallTraceNode>>,
```
Comment (DaniPopes) [inline src/tracing/builder/geth.rs:23]: should be [CallTraceNode]>

File: Cargo.toml
Comment (mattsse) [inline Cargo.toml:1]: could you submit this as a separate pr?

File: src/tracing/builder/geth.rs [hunk 2]
```diff
@@ -11,20 +11,26 @@ use alloy_rpc_types::trace::geth::{
...
     /// How the traces were recorded
     _config: TracingInspectorConfig,
 }
 
-impl GethTraceBuilder {
+impl<'a> GethTraceBuilder<'a> {
     /// Returns a new instance of the builder
-    pub fn new(nodes: Vec<CallTraceNode>, _config: TracingInspectorConfig) -> Self {
+    pub fn new(
```
Comment (mattsse) [inline src/tracing/builder/geth.rs:30]: I'd like to keep this function as is and add a new one `borrowed` instead

File: src/tracing/builder/geth.rs [hunk 3]
```diff
@@ -11,20 +11,26 @@ use alloy_rpc_types::trace::geth::{
...
-use std::collections::{BTreeMap, HashMap, VecDeque};
+use std::{
+    borrow::Cow,
+    collections::{BTreeMap, HashMap, VecDeque},
+};
 
 /// A type for creating geth style traces
 #[derive(Clone, Debug)]
-pub struct GethTraceBuilder {
+pub struct GethTraceBuilder<'a> {
```
Comment (mattsse) [inline src/tracing/builder/geth.rs:21]: This should then also implement `ToOwned`

File: src/tracing/builder/geth.rs [hunk 4]
```diff
@@ -114,7 +120,7 @@ impl GethTraceBuilder {
     ///
     /// This expects the gas used and return value for the
     /// [ExecutionResult](revm::primitives::ExecutionResult) of the executed transaction.
-    pub fn geth_call_traces(&self, opts: CallConfig, gas_used: u64) -> CallFrame {
+    pub fn geth_call_traces(&self, opts: &CallConfig, gas_used: u64) -> CallFrame {
```
Comment (mattsse) [inline src/tracing/builder/geth.rs:123]: this can stay `CallConfig` because this type is copy

**Review by mattsse — CHANGES_REQUESTED** [review]
cool, some nits and suggestions

**Review by mattsse — APPROVED** [review]

**Pull request merged**